shared_model/backend/protobuf/impl/proposal.cpp

```diff
@@ -5,38 +5,58 @@
 
 #include "backend/protobuf/proposal.hpp"
 
+#include "backend/protobuf/transaction.hpp"
+
 namespace shared_model {
   namespace proto {
     using namespace interface::types;
 
-    Proposal::Proposal(Proposal &&o) noexcept
-        : NonCopyableProto(std::move(o.proto_)) {}
-
-    Proposal &Proposal::operator=(Proposal &&o) noexcept {
-      proto_ = std::move(o.proto_);
-
-      hash_.invalidate();
-      transactions_.invalidate();
-      blob_.invalidate();
-
-      return *this;
+    struct Proposal::Impl {
+      Impl(TransportType &&ref) : proto_(std::move(ref)) {}
+      Impl(const TransportType &ref) : proto_(ref) {}
+      Impl(Impl &&o) noexcept = delete;
```

Comment:
No need to delete move ctor and assignment, as the class is managed by unique_ptr.